wgengine/magicsock/magicsock.go

```diff
@@ -3126,11 +3127,37 @@ const (
 // a discoEndpoint. (The subject is the discoEndpoint.endpointState
 // map key)
 type endpointState struct {
-	// all fields guarded by discoEndpoint.mu:
-	lastPing    time.Time
+	// all fields guarded by discoEndpoint.mu
+
+	// lastPing is the last (outgoing) ping time.
+	lastPing time.Time
+
+	// lastGotPing, if non-zero, means that this was an endpoint
+	// that we learned about at runtime (from an incoming ping)
+	// and that is not in the network map. If so, we keep the time
+	// updated and use it to discard old candidates.
+	lastGotPing time.Time
```

Comment:
I don't see any checking of this at send time. Should we avoid sending to peers that have been gone for a while, even though they might end up sitting around for a long time?

Reply:
Yeah, guess so. It's pruned on netmap updates at least but I can do better. One sec.